From 4058cec94e9264b7422ba1ceb3f39736e0ce971f Mon Sep 17 00:00:00 2001
From: sebastien baudouin <sebastien.baudouin@windriver.com>
Date: Tue, 28 Apr 2015 16:54:26 +0200
Subject: [PATCH 5/6] Launcher2: Make Graphic working on Qemu machine

Those modifications are in theory not needed but unfortunately without it the
graphic is not working.
---
 content/ContentModel.qml    |  2 +-
 content/ListItem.qml        |  2 +-
 content/NMapEffect.qml      |  2 +-
 content/NMapLightSource.qml |  2 +-
 content/PowerOffItem.qml    |  2 +-
 gdp-hmi-launcher2.qml       |  2 +-
 main.cpp                    | 12 ++++++++++++
 7 files changed, 18 insertions(+), 6 deletions(-)

diff --git a/content/ContentModel.qml b/content/ContentModel.qml
index 04a895e..6680509 100755
--- a/content/ContentModel.qml
+++ b/content/ContentModel.qml
@@ -14,7 +14,7 @@
  *                                  launching application
  *
  */
-import QtQuick 2.0
+import QtQuick 2.1
 
 ListModel {
     ListElement {
diff --git a/content/ListItem.qml b/content/ListItem.qml
index 4dcc978..2ff531e 100755
--- a/content/ListItem.qml
+++ b/content/ListItem.qml
@@ -13,7 +13,7 @@
  *                                  associated to the selected item
  *
  */
-import QtQuick 2.0
+import QtQuick 2.1
 
 Item {
     id: root
diff --git a/content/NMapEffect.qml b/content/NMapEffect.qml
index 05694e6..e7fbf06 100755
--- a/content/NMapEffect.qml
+++ b/content/NMapEffect.qml
@@ -11,7 +11,7 @@
  * List of changes:
  *
  */
-import QtQuick 2.0
+import QtQuick 2.1
 
 /* Simple normal mapping shader */
 
diff --git a/content/NMapLightSource.qml b/content/NMapLightSource.qml
index b0c787f..5b46ba4 100755
--- a/content/NMapLightSource.qml
+++ b/content/NMapLightSource.qml
@@ -11,7 +11,7 @@
  * List of changes:
  *
  */
-import QtQuick 2.0
+import QtQuick 2.1
 
 /* Light source for the normal mapping shader */
 
diff --git a/content/PowerOffItem.qml b/content/PowerOffItem.qml
index ea198e4..45b5025 100755
--- a/content/PowerOffItem.qml
+++ b/content/PowerOffItem.qml
@@ -15,7 +15,7 @@
  * 16.Mar.2015, Sebastien Baudouin, written based on template created by QtCreator
  *
  */
-import QtQuick 2.0
+import QtQuick 2.1
 
 Item {
     id: root
diff --git a/gdp-hmi-launcher2.qml b/gdp-hmi-launcher2.qml
index f9a58af..aeb446b 100755
--- a/gdp-hmi-launcher2.qml
+++ b/gdp-hmi-launcher2.qml
@@ -14,7 +14,7 @@
  *                                  stored and add a new item for powerOff
  *
  */
-import QtQuick 2.0
+import QtQuick 2.1
 import QtGraphicalEffects 1.0
 import "content"
 
diff --git a/main.cpp b/main.cpp
index 8a399bd..7961ad3 100755
--- a/main.cpp
+++ b/main.cpp
@@ -19,6 +19,12 @@
 #include <QtDBus>
 #include <QQmlContext>
 
+//For QEMU
+#include <QScreen>
+#include <QQuickWindow>
+#include <QQuickItem>
+
+
 #ifdef USE_DLT
 #include <dlt/dlt.h>
 DLT_DECLARE_CONTEXT(launcherTraceCtx);
@@ -73,6 +79,12 @@ int main(int argc, char* argv[])
     //let's show the view
     view.showFullScreen();
 
+    //For QEMU
+    QSize size = app.primaryScreen()->size();
+    view.rootObject()->setWidth(size.width());
+    view.rootObject()->setHeight(size.height());
+    view.show();
+
     //let's start the App
     ret = app.exec();
 
-- 
1.9.1

